internal/pkg/cli/cmd/cmd.go

```diff
@@ -0,0 +1,232 @@
+	root.cmd.SetOut(stdout)
+	root.cmd.SetErr(stderr)
+
+	// Setup templates
+	root.cmd.SetVersionTemplate("{{.Version}}")
+	root.cmd.SetUsageTemplate(
+		regexp.MustCompile(`Usage:(.|\n)*Aliases:`).ReplaceAllString(root.cmd.UsageTemplate(), usageTemplate),
+	)
+
+	// Persistent flags for all sub-commands
+	flags := root.cmd.PersistentFlags()
+	flags.SortFlags = true
+	flags.BoolP("help", "h", false, "print help for command")
+	flags.StringP("log-file", "l", "", "path to a log file for details")
+	flags.StringP("working-dir", "d", "", "use other working directory")
+	flags.StringP("storage-api-token", "t", "", "storage API token from your project")
+	flags.BoolP("verbose", "v", false, "print details")
+	flags.BoolP("verbose-api", "", false, "log each API request and response")
+
+	// Root command flags
+	root.cmd.Flags().SortFlags = true
+	root.cmd.Flags().BoolP("version", "V", false, "print version")
+
+	// Init when flags are parsed
+	root.cmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
+		// Temporary logger
+		tmpLogger := zap.NewNop().Sugar()
+
+		// Create filesystem abstraction
+		workingDir, _ := cmd.Flags().GetString(`working-dir`)
+		fs, err := fsFactory(tmpLogger, workingDir)
+		if err != nil {
+			return err
+		}
+
+		// Load values from flags and envs
+		if err = root.Options.Load(tmpLogger, envs, fs, cmd.Flags()); err != nil {
+			return err
+		}
+
+		// Setup logger
+		root.setupLogger()
+		fs.SetLogger(root.Logger)
+		root.Logger.Debug(`Working dir: `, filesystem.Join(fs.BasePath(), fs.WorkingDir()))
+
+		// Create dependencies container
+		root.Deps = dependencies.NewContainer(root.cmd.Context(), envs, fs, dialog.New(prompt), root.Logger, root.Options)
+
+		// Check version
+		if err := versionCheck.Run(root.Deps); err != nil {
+			// Ignore error, send to logs
+			root.Logger.Debugf(`Version check: %s.`, err.Error())
+		}
+
+		return nil
+	}
+
+	// Sub-commands
+	root.cmd.AddCommand(
+		StatusCommand(root),
+		InitCommand(root),
+		ValidateCommand(root),
+		PullCommand(root),
+		PushCommand(root),
+		DiffCommand(root),
+		PersistCommand(root),
+		FixPathsCommand(root),
+		EncryptCommand(root),
+		WorkflowsCommand(root),
+		CreateCommand(root),
+	)
+
+	return root
+}
+
+// Execute command or sub-command.
+func (root *RootCommand) Execute() (exitCode int) {
+	defer func() {
+		exitCode = root.tearDown(exitCode)
+	}()
+
+	// Logger can be nil, if error occurred before initialization
+	if root.Logger == nil {
+		root.setupLogger()
+	}
+
+	if err := root.cmd.Execute(); err != nil {
+		root.printError(err)
+		return 1
+	}
+	return 0
+}
+
+func (root *RootCommand) printError(errRaw error) {
+	// Convert to MultiError
+	err := utils.NewMultiError()
+	if !errors.As(errRaw, &err) {
+		err.Append(errRaw)
+	}
+
+	// Iterate over errors and replace message if needed
+	modifiedErrs := utils.NewMultiError()
+	for _, err := range err.Errors {
+		switch {
+		case errors.Is(err, dependencies.ErrMetadataDirFound):
```

Comment:
Niektore chyby su specificke pre `CLI`, ... ak napr. chyba `host`, tak v `CLI` je potrebne pouzit nejaky `flag`, .... pre `webservice` to bude zase inak. 

To je teraz uz mozne, chyba sa odchyti podla typu a zmeni sa.